pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -1004,9 +1006,31 @@ func (pl *dynamicResources) lookupClassParameters(logger klog.Logger, class *res
 	return nil, statusUnschedulable(logger, fmt.Sprintf("generated class parameters for %s.%s %s not found", class.ParametersRef.Kind, class.ParametersRef.APIGroup, klog.KRef(class.Namespace, class.ParametersRef.Name)))
 }
 
-func (pl *dynamicResources) lookupClaimParameters(logger klog.Logger, claim *resourcev1alpha2.ResourceClaim) (*resourcev1alpha2.ResourceClaimParameters, *framework.Status) {
+func (pl *dynamicResources) lookupClaimParameters(logger klog.Logger, class *resourcev1alpha2.ResourceClass, claim *resourcev1alpha2.ResourceClaim) (*resourcev1alpha2.ResourceClaimParameters, *framework.Status) {
+	defaultClaimParameters := resourcev1alpha2.ResourceClaimParameters{
+		Shareable: true,
+		DriverRequests: []resourcev1alpha2.DriverRequests{
+			{
+				DriverName: class.DriverName,
+				Requests: []resourcev1alpha2.ResourceRequest{
+					{
+						ResourceRequestModel: resourcev1alpha2.ResourceRequestModel{
+							// TODO: This only works because NamedResources is
+							// the only model currently implemented. We need to
+							// match the default to how the resources of this
+							// class are being advertized in a ResourceSlice.
+							NamedResources: &resourcev1alpha2.NamedResourcesRequest{
+								Selector: "true",
```

Comment:
I don't like that we would have to parse this selector. I would imagine that a case of "no parameters" is fairly common, so we should optimize for by understanding `nil`.

Reply:
The problem is that the scheduler doesn't inherently know which resource request model a driver is using to advertise its resources. We only know this by parsing this object and seeing which field is set (in this case `NamedResources`).

As such, I think a better approach would be to allow a cluster admin to define a default ResourceClaimParameters object and link it to a resource class. However, this is a larger chunk of work (and we are past the code freeze), so I proposed adding this PR as a stop-gap for 1.30.

Reply:
> The problem is that the scheduler doesn't inherently know which resource request model a driver is using.

Does it matter? The point is that it passes the "filter", regardless of the model.

Reply:
I'll let @pohly comment on that as I'm not intimately familiar with all of the code in this plugin. All I know is *some* default should be set and the default should allow sharing. Whether that's through an explicit object or a branch off of detecting `nil` I'll leave up to Patrick.

Reply:
I like the approach in this PR better because the rest of the code currently expects to determine the model based and thus the resources based on the parameters. Making it handle nil adds more special cases.

There's also a clear path towards what we really want (configurable default parameters). We need those because detecting the model may not be enough, we also need to figure out what the filters for a claim with no parameters should be. This depends on the driver and/or cluster, not the model.

https://github.com/kubernetes/kubernetes/issues/123858 is tracking this next step.